### ISSUE TITLE
Cannot subscribe by email

### ISSUE THREAD

**Issue by dest4** [docker-library/ghost#269]
Using ghost:4.20.4, visitors are not able to subscribe by email, as the step that sends a magic link seems to fail. Note that mailing works well when I publish articles, so my Mailgun API key seems to be properly setup. Only subscribing using public form seems to be broken.

![image](https://user-images.githubusercontent.com/9466937/139953797-e6fa5e3c-a435-48ac-bb19-1762596e520a.png)
(I have replaced the email address with a placeholder)

the modal shows the following error after 60 seconds
![image](https://user-images.githubusercontent.com/9466937/139953885-678eceb8-2a74-4e38-b811-40b63ef5332e.png)

in the dev console, the endpoint https://xxx.xxx/members/api/send-magic-link/ has a status 504 Gateway time out.
In the log, it says 200: 
![image](https://user-images.githubusercontent.com/9466937/139954085-000085bc-ce68-4039-b5ba-441f91a51877.png)

Note that when I use another email (namely, `foo@bar.com`), the error is different. The endpoint receives a 500 internal server error status and the logs show
![image](https://user-images.githubusercontent.com/9466937/139954198-1d518bfa-e33b-4f7e-ae3e-bf0e15ee8566.png)
![image](https://user-images.githubusercontent.com/9466937/139955032-5df0d339-b70a-43c6-94d9-18c4def85aac.png)

Therefore it seems Ghost and Mailgun do not communicate well anymore. The operations above do not create log lines in the Mailgun dashboard, where I only see logs related to recent publications.

**Comment by pascalandy** [comment]
Is it working using 4.20.3 or 4.19.x ?

**Comment by dest4** [comment]
It is not working with 4.20.3, with identical symptoms. I am a bit hesitant to test with 4.19.x, I fear downgrading could break the DB. But before submitting the ticket, I have updated from 4.16.0 and it was the same.
FWIW, I'm using EU Mailgun servers.

**Comment by pascalandy** [comment]
It feel like you should go to Ghost forums. It's unlikely an issue with the container.

**Comment by dest4** [comment]
Will do! Thanks for your support

**Comment by dest4** [comment]
Follow up: I did not realize that Mailgun transactional and bulk email settings are defined at different places. Transactional seems to be setup in `config.production.json` (see [doc](https://ghost.org/docs/config/#mail)) while bulk email API setup is done in the settings within ghost interface.

I did not properly configure the json. Now that I have done it, subscription works as intended. To do so, I have edited my docker command in the following way to have a proper json within the image.
```
sudo docker run -d --name blog-v4 -e url=https://my.domain.fr -p 3001:2368 \
  -v /var/lib/ghost/content:/var/lib/ghost/content \
  -v /var/lib/ghost/config.production.json:/var/lib/ghost/config.production.json ghost:4
```
